internal/testbed/integration/k8scombined/e2e_test.go

```diff
@@ -0,0 +1,429 @@
+      exporters: 
+        - otlphttp`,
+			fmt.Sprintf(`
+  otlphttp/metrics:
+    endpoint: http://%s:4320
+    headers:
+      Authorization: "Api-Token "
+  otlphttp/logs:
+    endpoint: http://%s:4319
+    headers:
+      Authorization: "Api-Token "
+
+service:
+  extensions:
+    - health_check
+  pipelines:
+    metrics:
+      receivers:
+        - k8s_cluster
+      processors:
+        - k8sattributes
+        - transform
+        - cumulativetodelta
+      exporters:
+        - otlphttp/metrics
+    logs:
+      receivers: 
+        - k8sobjects
+      processors:
+        - transform
+      exporters: 
+        - otlphttp/logs`, host, host),
+		},
+	})
+
+	require.NoErrorf(t, err, "Failed to read collector config from file %s", collectorConfigPath)
+	collectorObjs1 := otelk8stest.CreateCollectorObjects(
+		t,
+		k8sClient,
+		testID,
+		filepath.Join(testDir, "collector-gateway"),
+		map[string]string{
+			"ContainerRegistry": os.Getenv("CONTAINER_REGISTRY"),
+			"CollectorConfig":   collectorConfig,
+		},
+		host,
+	)
+
+	// create deployment
+	deploymentFile := filepath.Join(testDir, "testobjects", "deployment.yaml")
+	buf, err = os.ReadFile(deploymentFile)
+	require.NoErrorf(t, err, "failed to read deployment object file %s", deploymentFile)
+	deploymentObj, err := otelk8stest.CreateObject(k8sClient, buf)
+	require.NoErrorf(t, err, "failed to create k8s deployment from file %s", deploymentFile)
+
+	defer func() {
+		for _, obj := range append(collectorObjs1, deploymentObj) {
+			require.NoErrorf(t, otelk8stest.DeleteObject(k8sClient, obj), "failed to delete object %s", obj.GetName())
+		}
+	}()
+
+	t.Logf("Checking logs...")
+
+	expectedLogEvents := false
+	oteltest.WaitForLogs(t, 1, logsConsumer)
+
+	for _, r := range logsConsumer.AllLogs() {
+		for i := 0; i < r.ResourceLogs().Len(); i++ {
+			sm := r.ResourceLogs().At(i).ScopeLogs().At(0).LogRecords()
+			for j := 0; j < sm.Len(); j++ {
+				bodyMap := sm.At(j).Body().Map()
+				if kind, ok := bodyMap.Get("kind"); ok && kind.Str() == "Event" {
+					if _, ok := bodyMap.Get("message"); ok {
+						expectedLogEvents = true
+					}
+				}
+			}
+		}
+	}
+
+	require.True(t, expectedLogEvents, "Event logs not found")
+
+	t.Logf("Logs checked successfully")
+	t.Logf("Checking gateway metrics...")
+
+	oteltest.WaitForMetrics(t, 5, metricsConsumerGateway)
+
+	// the commented line below writes the received list of metrics to the expected.yaml
+	// require.Nil(t, golden.WriteMetrics(t, expectedGatewayFile, metricsConsumerGateway.AllMetrics()[len(metricsConsumerGateway.AllMetrics())-1]))
+
+	expected, err = golden.ReadMetrics(expectedGatewayFile)
+	require.NoError(t, err)
+
+	require.EventuallyWithT(t, func(tt *assert.CollectT) {
+		assert.NoError(tt, pmetrictest.CompareMetrics(expected, metricsConsumerGateway.AllMetrics()[len(metricsConsumerGateway.AllMetrics())-1],
+			defaultOptions...,
+		),
+		)
+	}, 3*time.Minute, 1*time.Second)
+
+	t.Logf("Gateway metrics checked successfully")
```

Comment:
if i understand the test correctly, both collectors (agent and gateway) are tested separately, as the otlp exporter of the agent is modified to send to the local sink instead of `otelcolsvc:4317`. This definitely makes sense, but I was wondering if we should also add a test where keep the agent collectors' exporter configuration unchanged to send data to the gateway, and then only check what we get from the gateway collector?
If yes, I think that should be a follow up though, so I would not include that in this PR

Reply:
that makes sense, thanks for elaborating!